mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerUtils.scala

```diff
@@ -50,6 +50,44 @@ trait MesosSchedulerUtils extends Logging {
   protected var mesosDriver: SchedulerDriver = null
 
   /**
+   * Returns the configured set of roles that an offer can be selected from
+   * @param conf Spark configuration
+   */
+  protected def getAcceptedResourceRoles(conf: SparkConf): Set[String] = {
+    getAcceptedResourceRoles(
+      conf.getBoolean("spark.mesos.ignoreDefaultRoleResources", false),
+      conf.getOption("spark.mesos.role"))
+  }
+  /**
+   * Returns the configured set of roles that an offer can be selected from
+   * @param props Mesos driver description schedulerProperties map
+   */
+  protected def getAcceptedResourceRoles(props: Map[String, String]): Set[String] = {
+    getAcceptedResourceRoles(
+      props.get("spark.mesos.ignoreDefaultRoleResources") match {
```

Comment:
This might be my taste only but I'm finding this style hard to read. 
`props.get(...).map(_.toBoolean).getOrElse(false)`? It's what I'd expect to see in the spark code base but don't know if it's objectively any better. Still I'd break this out into a `val` rather than squeeze this into a method arg